src/pkg/selectors/selectors.go

```diff
@@ -146,6 +148,46 @@ func (s Selector) DiscreteResourceOwners() []string {
 	return split(s.ResourceOwners.Target)
 }
 
+// SetDiscreteOwnerIDName ensures the selector has the correct discrete owner
+// id and name.  It is assumed that these values are sourced using the current
+// s.DiscreteOwner as input.  The reason for taking in both the id and name, and
+// not just the name, is so that constructors can input owner aliases in place
+// of  ids, with the expectation that the two will get sorted and re-written
+// later on with this setter.
+func (s Selector) SetDiscreteOwnerIDName(id, name string) Selector {
```

Comment:
It's to maintain interchangability with human readable names.  (also, aside, thinking about renaming "name" here to "alias" or "id-alias", because an item can have both an identifiable display name (webURL, principalName), and also a proper display name: the person's or the site's name).

Consider userIDs vs userPNs (or, for sharepoint, siteIDs vs webURLs).  If we want to accept both at the sdk/cli layer, then transformation of names to ids has to happen _before_ building the selector (which pushes GC upstream, and I don't want that), or I have to defer building the selector and push some "selector parameters" interface onto pkg/repository (also not wanted), _or_ allow building selectors with an owners list that can contain both names and ids (which is what we're doing).  The DiscreteOwner is set in the selector before we transform from ID to Name, therefore we need a way to set the values correctly after transformation.

Reply:
So I can see how this function could be useful, but I think the implementation is still confusing. To sum up the code below, the rules for it seem to be:
* if no id is given, ignore name and reuse the ID from the original selector as the name and the id
* if an id but no name is given, reuse the id for both the id and name
* if an id and name is given use them as denoted by their names

None of this is explained in the comment and the first case in particular seems like it would be a good footgun

Reply:
Fair.  Made two quick changes: added an explanation to the func header, and changed the behavior when given an empty id.  Now it'll retain the original owner value, and not clobber the new name.